src/adjoint/DASolver/DASolver.C

```diff
@@ -533,19 +531,25 @@ void DASolver::getForcesInternal(
     volumeForceField.write();
 
     // The above volumeForceField is face-centered, we need to interpolate it to point-centered
-    label pointCounter = 0;
-    List<label> globalIndex = pointList.clone();
-    globalIndex = -1;
-
     pointField meshPoints = meshPtr_->points();
 
     vector nodeForce(vector::zero);
 
+    label patchStart = 0;
     forAll(patchListSort, cI)
     {
         // get the patch id label
         label patchI = meshPtr_->boundaryMesh().findPatchID(patchListSort[cI]);
+        label patchIPoints = boundaryMesh.findPatchID(patchListSort[cI]);
+
+        label nPointsPatch = boundaryMesh[patchIPoints].size();
+        List<scalar> fXTemp(nPointsPatch);
+        List<scalar> fYTemp(nPointsPatch);
+        List<scalar> fZTemp(nPointsPatch);
+        List<label> pointListTemp(nPointsPatch);
```

Comment:
Recreating these arrays is the simplest way to have the temporary force arrays with the correct length for each patch. Is this too computationally expensive or will it lead to any memory issues?